packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -215,16 +220,96 @@ abstract contract PermissionManager is Initializable {
-            _isGranted(_where, ANY_ADDR, _permissionId, _data) || // check if anyone has permission for `_permissionId` on `_where`
-            _isGranted(ANY_ADDR, _who, _permissionId, _data); // check if `_who` has permission for `_permissionI` on any contract
+        // Specific caller and target
+        {
+            address specificPermissionOrCondition = permissionsHashed[
+                permissionHash({_where: _where, _who: _who, _permissionId: _permissionId})
+            ];
+
+            if (specificPermissionOrCondition == ALLOW_FLAG) return true;
+
+            if (specificPermissionOrCondition != UNSET_FLAG) {
+                return
+                    _checkCondition({
+                        _condition: specificPermissionOrCondition,
+                        _where: _where,
+                        _who: _who,
+                        _permissionId: _permissionId,
+                        _data: _data
+                    });
+            }
+        }
+
+        // Generic caller `_who: ANY_ADDR`  (only possible by granting with `grantWithCondition`)
+        {
+            address genericCallerCondition = permissionsHashed[
+                permissionHash({_where: _where, _who: ANY_ADDR, _permissionId: _permissionId})
+            ];
+
+            if (genericCallerCondition != UNSET_FLAG) {
+                return
+                    _checkCondition({
+                        _condition: genericCallerCondition,
+                        _where: _where,
+                        _who: _who,
+                        _permissionId: _permissionId,
+                        _data: _data
+                    });
+            }
+        }
+
+        // Generic target `_where: ANY_ADDR` (only possible by granting with `grantWithCondition`)
+        {
+            address genericTargetCondition = permissionsHashed[
+                permissionHash({_where: ANY_ADDR, _who: _who, _permissionId: _permissionId})
+            ];
+            if (genericTargetCondition != UNSET_FLAG) {
+                return
+                    _checkCondition({
+                        _condition: genericTargetCondition,
+                        _where: _where,
+                        _who: _who,
+                        _permissionId: _permissionId,
+                        _data: _data
+                    });
+            }
+        }
+
+        return false;
+    }
+
+    /// @notice Checks an condition contract by doing an external call.
+    /// @param _condition The condition contract that is asked.
+    /// @param _where The address of the target contract for which `_who` receives permission.
+    /// @param _who The address (EOA or contract) owning the permission.
+    /// @param _permissionId The permission identifier.
+    /// @param _data The optional data passed to the `PermissionCondition` registered.
+    /// @return Returns true if `_who` has the permissions on the contract via the specified permissionId identifier.
```

Comment:
```suggestion
    /// @return Returns true if `_who` has the permissions on the contract (`_where`) via the specified permission  identifier (`_permissionId`) with the optional data (`_data`) passed.
```

Reply:
Improved in ab2a4203454ba16ac0d10285723b51055d619afe